examples/functions/golang/backfill/mmf/matchfunction.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mmf
+
+import (
+	"fmt"
+	"time"
+
+	"log"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"google.golang.org/grpc"
+	"open-match.dev/open-match/pkg/matchfunction"
+	"open-match.dev/open-match/pkg/pb"
+)
+
+const (
+	playersPerMatch = 2
+	openSlotsKey    = "open-slots"
+	matchName       = "backfill-matchfunction"
+)
+
+type matchFunctionService struct {
+	grpc               *grpc.Server
+	queryServiceClient pb.QueryServiceClient
+	port               int
+}
+
```

Comment:
Given this is an example, I would strongly recommend adding comment block on the top giving an overview of whats the behavior of this method. How does it return matches, when does it return backfill matches, when does it populate more tickets into a backfill etc.?

It would make it much simpler to review if you also add small comments on top of each of the helpers - so it is easy to understand how they piece together.